File: easy/find-nearest-point-that-has-the-same-x-or-y-coordinate/find-nearest-point-that-has-the-same-x-or-y-coordinate.c
#include <limits.h>

int nearestValidPoint(int x, int y, int **points, int pointsSize, int *pointsColSize)
{
    int minDistance = INT_MAX;
    int minDistanceIndex = -1;
    for (int i = 0; i < pointsSize; i++)
    {
        int x1 = points[i][0];
        int y1 = points[i][1];
        if (x1 != x && y1 != y)
        {
            continue;
        }
        int distance = abs(x - x1) + abs(y - y1);
        if (distance < minDistance)
        {
            minDistance = distance;
            minDistanceIndex = i;
        }
    }
    return minDistanceIndex;
}

File: easy/find-nearest-point-that-has-the-same-x-or-y-coordinate/find-nearest-point-that-has-the-same-x-or-y-coordinate.py
class Solution:
    def nearestValidPoint(self, x: int, y: int, points: List[List[int]]) -> int:
        minDistance = sys.maxsize
        minDistanceIndex = -1
        for index, point in enumerate(points):
            x1 = points[index][0]
            y1 = points[index][1]
            if x1 != x and y1 != y:
                continue
            distance = abs(x - x1) + abs(y - y1)
            if distance < minDistance:
                minDistance = distance
                minDistanceIndex = index
        return minDistanceIndex
